src/callbacks_step/euler_acoustics_coupling.jl

```diff
@@ -0,0 +1,205 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+
+
+@doc raw"""
+    EulerAcousticsCouplingCallback
+
+!!! warning "Experimental code"
+    This callback is experimental and may change in any future release.
+
+A callback that couples the acoustic perturbation equations and compressible Euler equations. Must
+be used in conjunction with [`SemidiscretizationEulerAcoustics`](@ref).
+This callback manages the flow solver - which is always one time step ahead of the
+acoustics solver - and calculates the acoustic source term after each time step. The linearized
+Lamb vector is used as the source term, i.e.
+```math
+\mathbf{s} = -(\mathbf{\omega'} \times \bar{\mathbf{v}}
+  + \bar{\mathbf{\omega}} \times \mathbf{v'}),
+```
+where ``\mathbf{v}`` denotes the velocity, ``\mathbf{\omega}`` denotes the vorticity, the bar
+``\bar{(\cdot)}`` indicates time-averaged quantities (see [`AveragingCallback`](@ref)) and prime
+``(\cdot)'`` denotes perturbed quantities defined by ``\phi' = \phi - \bar{\phi}``. Note that
+the perturbed quantities here are entirely based on the pure flow solution and should not be
+confused with the state variables of the acoustic perturbation equations.
+
+In addition, this callback manages the step size for both solvers
+and initializes the mean values of the acoustic perturbation equations using results obtained with
+the [`AveragingCallback`](@ref).
+
+- Michael Schlottke-Lakemper (2017)
+  A direct-hybrid method for aeroacoustic analysis
+  [DOI: 10.18154/RWTH-2017-04082](https://doi.org/10.18154/RWTH-2017-04082)
+"""
+mutable struct EulerAcousticsCouplingCallback{RealT<:Real, MeanValues, IntegratorEuler}
+  stepsize_callback_acoustics::StepsizeCallback{RealT}
+  stepsize_callback_euler::StepsizeCallback{RealT}
+  mean_values::MeanValues
+  integrator_euler::IntegratorEuler
+end
```

Comment:
@andrewwinters5000 Thanks for your review! Regarding your [comment](https://github.com/trixi-framework/Trixi.jl/pull/712#pullrequestreview-753890742): I updated the docstring to include the formula for the acoustic source and added a reference. I've chosen the `EulerAcousticsCouplingCallback` docstring because that's where the source term is calculated.

Reply:
Thanks! This is a very helpful and descriptive docstring